### PR TITLE
Fix a NRE in VB query expression binding

### LINKED ISSUE #1099: VS crash: NRE from VB binder while typing complex LINQ query
Repro:

```vb
Class C
    Shared Function Id(Of T)(a As T, i As Integer) As T
        Return a
    End Function

    Sub F2()
        Dim result = From a In Id({1}, 1), b In Id({1, 2}, 2)
                     From c In Id({1, 2, 3}, 3)
                     Let d = Id(1, 4), e = Id(2, 5)
                     Distinct
                     Aggregate f In Id({1}, 6), g In Id({2}, 7)
                         From j In Id({1}, 9)
                         Let h = Id(1, 4), i = Id(2, 5)
                         Where Id(g < 2, 8)
                     Into Count(), Distinct()

    End Sub
End Class
```

Start typing "Take While" clause after "Distinct".

Crash dump:

%internal_share%\tomat\Bugs\1099

Stack trace:

```
>	Microsoft.CodeAnalysis.VisualBasic.dll!Microsoft.CodeAnalysis.VisualBasic.Binder.CompleteAggregateClauseBinding(Microsoft.CodeAnalysis.VisualBasic.Syntax.AggregateClauseSyntax aggregate, Microsoft.CodeAnalysis.SyntaxList(Of Microsoft.CodeAnalysis.VisualBasic.Syntax.QueryClauseSyntax).Enumerator operatorsEnumerator, System.Collections.Immutable.ImmutableArray(Of Microsoft.CodeAnalysis.VisualBasic.Symbols.RangeVariableSymbol) sourceRangeVariablesPart1, System.Collections.Immutable.ImmutableArray(Of Microsoft.CodeAnalysis.VisualBasic.Symbols.RangeVariableSymbol) sourceRangeVariablesPart2, Microsoft.CodeAnalysis.VisualBasic.BoundExpression firstSelectExpression, Microsoft.CodeAnalysis.VisualBasic.Binder.QueryLambdaBinder firstSelectSelectorBinder, System.Collections.Immutable.ImmutableArray(Of Microsoft.CodeAnalysis.VisualBasic.Symbols.RangeVariableSymbol) firstSelectDeclaredRangeVariables, Microsoft.CodeAnalysis.VisualBasic.Symbols.TypeSymbol firstSelectCompoundVariableType, Microsoft.CodeAnalysis.VisualBasic.BoundQueryClauseBase group, Microsoft.CodeAnalysis.VisualBasic.Binder.IntoClauseDisallowGroupReferenceBinder intoBinder, Microsoft.CodeAnalysis.DiagnosticBag diagnostics) Line 406	Basic
 	Microsoft.CodeAnalysis.VisualBasic.dll!Microsoft.CodeAnalysis.VisualBasic.Binder.BindAggregateClause(Microsoft.CodeAnalysis.VisualBasic.BoundQueryClauseBase source, Microsoft.CodeAnalysis.VisualBasic.Syntax.AggregateClauseSyntax aggregate, Microsoft.CodeAnalysis.SyntaxList(Of Microsoft.CodeAnalysis.VisualBasic.Syntax.QueryClauseSyntax).Enumerator operatorsEnumerator, Microsoft.CodeAnalysis.DiagnosticBag diagnostics) Line 352	Basic
 	Microsoft.CodeAnalysis.VisualBasic.dll!Microsoft.CodeAnalysis.VisualBasic.Binder.BindSubsequentQueryOperators(Microsoft.CodeAnalysis.VisualBasic.BoundQueryClauseBase source, Microsoft.CodeAnalysis.SyntaxList(Of Microsoft.CodeAnalysis.VisualBasic.Syntax.QueryClauseSyntax).Enumerator operators, Microsoft.CodeAnalysis.DiagnosticBag diagnostics) Line 136	Basic
 	Microsoft.CodeAnalysis.VisualBasic.dll!Microsoft.CodeAnalysis.VisualBasic.Binder.BindFromQueryExpression(Microsoft.CodeAnalysis.VisualBasic.Syntax.QueryExpressionSyntax query, Microsoft.CodeAnalysis.SyntaxList(Of Microsoft.CodeAnalysis.VisualBasic.Syntax.QueryClauseSyntax).Enumerator operators, Microsoft.CodeAnalysis.DiagnosticBag diagnostics) Line 159	Basic
 	Microsoft.CodeAnalysis.VisualBasic.dll!Microsoft.CodeAnalysis.VisualBasic.Binder.BindQueryExpression(Microsoft.CodeAnalysis.VisualBasic.Syntax.QueryExpressionSyntax query, Microsoft.CodeAnalysis.DiagnosticBag diagnostics) Line 58	Basic
 	Microsoft.CodeAnalysis.VisualBasic.dll!Microsoft.CodeAnalysis.VisualBasic.Binder.BindExpression(Microsoft.CodeAnalysis.VisualBasic.Syntax.ExpressionSyntax node, Boolean isInvocationOrAddressOf, Boolean isOperandOfConditionalBranch, Boolean eventContext, Microsoft.CodeAnalysis.DiagnosticBag diagnostics) Line 185	Basic
 	Microsoft.CodeAnalysis.VisualBasic.dll!Microsoft.CodeAnalysis.VisualBasic.Binder.ComputeVariableType(Microsoft.CodeAnalysis.VisualBasic.Symbols.LocalSymbol symbol, Microsoft.CodeAnalysis.VisualBasic.Syntax.ModifiedIdentifierSyntax modifiedIdentifierOpt, Microsoft.CodeAnalysis.VisualBasic.Syntax.AsClauseSyntax asClauseOpt, Microsoft.CodeAnalysis.VisualBasic.Syntax.EqualsValueSyntax equalsValueOpt, Microsoft.CodeAnalysis.VisualBasic.BoundExpression valueExpression, Microsoft.CodeAnalysis.VisualBasic.Symbols.TypeSymbol asClauseType, Microsoft.CodeAnalysis.DiagnosticBag diagnostics) Line 1298	Basic
 	Microsoft.CodeAnalysis.VisualBasic.dll!Microsoft.CodeAnalysis.VisualBasic.Binder.BindVariableDeclaration(Microsoft.CodeAnalysis.VisualBasic.VisualBasicSyntaxNode tree, Microsoft.CodeAnalysis.VisualBasic.Syntax.ModifiedIdentifierSyntax name, Microsoft.CodeAnalysis.VisualBasic.Syntax.AsClauseSyntax asClauseOpt, Microsoft.CodeAnalysis.VisualBasic.Syntax.EqualsValueSyntax equalsValueOpt, Microsoft.CodeAnalysis.DiagnosticBag diagnostics, Boolean skipAsNewInitializer) Line 1063	Basic
 	Microsoft.CodeAnalysis.VisualBasic.dll!Microsoft.CodeAnalysis.VisualBasic.Binder.BindVariableDeclarators(Microsoft.CodeAnalysis.SeparatedSyntaxList(Of Microsoft.CodeAnalysis.VisualBasic.Syntax.VariableDeclaratorSyntax) declarators, Microsoft.CodeAnalysis.DiagnosticBag diagnostics) Line 1006	Basic
 	Microsoft.CodeAnalysis.VisualBasic.dll!Microsoft.CodeAnalysis.VisualBasic.Binder.BindLocalDeclaration(Microsoft.CodeAnalysis.VisualBasic.Syntax.LocalDeclarationStatementSyntax node, Microsoft.CodeAnalysis.DiagnosticBag diagnostics) Line 944	Basic
 	Microsoft.CodeAnalysis.VisualBasic.dll!Microsoft.CodeAnalysis.VisualBasic.Binder.BindStatement(Microsoft.CodeAnalysis.VisualBasic.Syntax.StatementSyntax node, Microsoft.CodeAnalysis.DiagnosticBag diagnostics) Line 74	Basic
 	Microsoft.CodeAnalysis.VisualBasic.dll!Microsoft.CodeAnalysis.VisualBasic.MemberSemanticModel.IncrementalBinder.BindStatement(Microsoft.CodeAnalysis.VisualBasic.Syntax.StatementSyntax node, Microsoft.CodeAnalysis.DiagnosticBag diagnostics) Line 1968	Basic
 	Microsoft.CodeAnalysis.VisualBasic.dll!Microsoft.CodeAnalysis.VisualBasic.VBSemanticModel.Bind(Microsoft.CodeAnalysis.VisualBasic.Binder binder, Microsoft.CodeAnalysis.VisualBasic.VisualBasicSyntaxNode node, Microsoft.CodeAnalysis.DiagnosticBag diagnostics) Line 1592	Basic
 	Microsoft.CodeAnalysis.VisualBasic.dll!Microsoft.CodeAnalysis.VisualBasic.MemberSemanticModel.GuardedIncrementalBind(Microsoft.CodeAnalysis.VisualBasic.VisualBasicSyntaxNode bindingRoot, Microsoft.CodeAnalysis.VisualBasic.Binder enclosingBinder) Line 1842	Basic
 	Microsoft.CodeAnalysis.VisualBasic.dll!Microsoft.CodeAnalysis.VisualBasic.MemberSemanticModel.GetBoundNodes(Microsoft.CodeAnalysis.VisualBasic.VisualBasicSyntaxNode node) Line 1762	Basic
 	Microsoft.CodeAnalysis.VisualBasic.dll!Microsoft.CodeAnalysis.VisualBasic.MemberSemanticModel.GetUpperBoundNode(Microsoft.CodeAnalysis.VisualBasic.VisualBasicSyntaxNode node) Line 141	Basic
 	Microsoft.CodeAnalysis.VisualBasic.dll!Microsoft.CodeAnalysis.VisualBasic.MemberSemanticModel.GetBoundNodeSummary(Microsoft.CodeAnalysis.VisualBasic.VisualBasicSyntaxNode node) Line 199	Basic
 	Microsoft.CodeAnalysis.VisualBasic.dll!Microsoft.CodeAnalysis.VisualBasic.MemberSemanticModel.GetExpressionSymbolInfo(Microsoft.CodeAnalysis.VisualBasic.Syntax.ExpressionSyntax node, Microsoft.CodeAnalysis.VisualBasic.VBSemanticModel.SymbolInfoOptions options, System.Threading.CancellationToken cancellationToken) Line 778	Basic
 	Microsoft.CodeAnalysis.VisualBasic.dll!Microsoft.CodeAnalysis.VisualBasic.VBSemanticModel.GetSymbolInfo(Microsoft.CodeAnalysis.VisualBasic.Syntax.ExpressionSyntax expression, System.Threading.CancellationToken cancellationToken) Line 154	Basic
 	Microsoft.CodeAnalysis.VisualBasic.dll!Microsoft.CodeAnalysis.VisualBasic.VisualBasicExtensions.GetSymbolInfo(Microsoft.CodeAnalysis.SemanticModel semanticModel, Microsoft.CodeAnalysis.VisualBasic.Syntax.ExpressionSyntax expression, System.Threading.CancellationToken cancellationToken) Line 1266	Basic
 	Microsoft.CodeAnalysis.VisualBasic.Workspaces.dll!Microsoft.CodeAnalysis.VisualBasic.Classification.Classifiers.NameSyntaxClassifier.ClassifyNameSyntax(Microsoft.CodeAnalysis.VisualBasic.Syntax.NameSyntax node, Microsoft.CodeAnalysis.SemanticModel semanticModel, System.Threading.CancellationToken cancellationToken) Line 48	Basic
 	Microsoft.CodeAnalysis.VisualBasic.Workspaces.dll!Microsoft.CodeAnalysis.VisualBasic.Classification.Classifiers.NameSyntaxClassifier.ClassifyNode(Microsoft.CodeAnalysis.SyntaxNode syntax, Microsoft.CodeAnalysis.SemanticModel semanticModel, System.Threading.CancellationToken cancellationToken) Line 26	Basic
 	Microsoft.CodeAnalysis.Workspaces.dll!Microsoft.CodeAnalysis.Classification.AbstractClassificationService.Worker.ClassifyNode(Microsoft.CodeAnalysis.SyntaxNode syntax) Line 121	C#
 	Microsoft.CodeAnalysis.Workspaces.dll!Microsoft.CodeAnalysis.Classification.AbstractClassificationService.Worker.ProcessNodes() Line 100	C#
 	Microsoft.CodeAnalysis.Workspaces.dll!Microsoft.CodeAnalysis.Classification.AbstractClassificationService.Worker.Classify(Microsoft.CodeAnalysis.Workspace workspace, Microsoft.CodeAnalysis.SemanticModel semanticModel, Microsoft.CodeAnalysis.Text.TextSpan textSpan, System.Collections.Generic.List<Microsoft.CodeAnalysis.Classification.ClassifiedSpan> list, System.Func<Microsoft.CodeAnalysis.SyntaxNode, System.Collections.Generic.List<Microsoft.CodeAnalysis.Classification.Classifiers.ISyntaxClassifier>> getNodeClassifiers, System.Func<Microsoft.CodeAnalysis.SyntaxToken, System.Collections.Generic.List<Microsoft.CodeAnalysis.Classification.Classifiers.ISyntaxClassifier>> getTokenClassifiers, System.Threading.CancellationToken cancellationToken) Line 63	C#
 	Microsoft.CodeAnalysis.Workspaces.dll!Microsoft.CodeAnalysis.Classification.AbstractClassificationService.AddSemanticClassificationsAsync(Microsoft.CodeAnalysis.Document document, Microsoft.CodeAnalysis.Text.TextSpan textSpan, System.Func<Microsoft.CodeAnalysis.SyntaxNode, System.Collections.Generic.List<Microsoft.CodeAnalysis.Classification.Classifiers.ISyntaxClassifier>> getNodeClassifiers, System.Func<Microsoft.CodeAnalysis.SyntaxToken, System.Collections.Generic.List<Microsoft.CodeAnalysis.Classification.Classifiers.ISyntaxClassifier>> getTokenClassifiers, System.Collections.Generic.List<Microsoft.CodeAnalysis.Classification.ClassifiedSpan> result, System.Threading.CancellationToken cancellationToken) Line 32	C#
 	Microsoft.CodeAnalysis.EditorFeatures.dll!Microsoft.CodeAnalysis.Editor.Implementation.Classification.AbstractEditorClassificationService.AddSemanticClassificationsAsync(Microsoft.CodeAnalysis.Document document, Microsoft.CodeAnalysis.Text.TextSpan textSpan, System.Collections.Generic.List<Microsoft.CodeAnalysis.Classification.ClassifiedSpan> result, System.Threading.CancellationToken cancellationToken) Line 31	C#
 	Microsoft.CodeAnalysis.EditorFeatures.dll!Microsoft.CodeAnalysis.Editor.Implementation.Classification.SemanticClassificationTaggerProvider.TagProducer.ProduceTagsAsync(Microsoft.CodeAnalysis.Document document, Microsoft.VisualStudio.Text.SnapshotSpan snapshotSpan, int? caretPosition, System.Threading.CancellationToken cancellationToken) Line 65	C#
 	Microsoft.CodeAnalysis.EditorFeatures.dll!Microsoft.CodeAnalysis.Editor.Shared.Tagging.AbstractSingleDocumentTagProducer<Microsoft.VisualStudio.Text.Tagging.IClassificationTag>.ProduceTagsAsync(System.Collections.Generic.IEnumerable<Microsoft.CodeAnalysis.Editor.DocumentSnapshotSpan> snapshotSpans, Microsoft.VisualStudio.Text.SnapshotPoint? caretPosition, System.Threading.CancellationToken cancellationToken) Line 45	C#
 	Microsoft.CodeAnalysis.EditorFeatures.dll!Microsoft.CodeAnalysis.Editor.Shared.Tagging.ProducerPopulatedTagSource<Microsoft.VisualStudio.Text.Tagging.IClassificationTag>.RecomputeTagsAsync(Microsoft.VisualStudio.Text.SnapshotPoint? caretPosition, Microsoft.CodeAnalysis.Text.TextChangeRange? textChangeRange, System.Collections.Generic.IEnumerable<Microsoft.CodeAnalysis.Editor.DocumentSnapshotSpan> spansToCompute, System.Threading.CancellationToken cancellationToken) Line 537	C#
 	Microsoft.CodeAnalysis.EditorFeatures.dll!Microsoft.CodeAnalysis.Editor.Shared.Tagging.TagSources.SemanticBufferTagSource<System.__Canon>.<>n__0(Microsoft.VisualStudio.Text.SnapshotPoint? caretPosition, Microsoft.CodeAnalysis.Text.TextChangeRange? textChangeRange, System.Collections.Generic.IEnumerable<Microsoft.CodeAnalysis.Editor.DocumentSnapshotSpan> spansToCompute, System.Threading.CancellationToken cancellationToken)	Unknown
 	Microsoft.CodeAnalysis.EditorFeatures.dll!Microsoft.CodeAnalysis.Editor.Shared.Tagging.TagSources.SemanticBufferTagSource<Microsoft.VisualStudio.Text.Tagging.IClassificationTag>.RecomputeTagsAsync(Microsoft.CodeAnalysis.Text.TextChangeRange? range, System.Collections.Generic.IEnumerable<Microsoft.CodeAnalysis.Editor.DocumentSnapshotSpan> spansToCompute, Microsoft.CodeAnalysis.Document document, Microsoft.VisualStudio.Text.ITextSnapshot snapshot, Microsoft.CodeAnalysis.VersionStamp oldVersion, Microsoft.CodeAnalysis.VersionStamp newVersion, System.Threading.CancellationToken cancellationToken) Line 129	C#
 	Microsoft.CodeAnalysis.EditorFeatures.dll!Microsoft.CodeAnalysis.Editor.Shared.Tagging.TagSources.SemanticBufferTagSource<Microsoft.VisualStudio.Text.Tagging.IClassificationTag>.RecomputeTagsAsync(Microsoft.VisualStudio.Text.SnapshotPoint? caret, Microsoft.CodeAnalysis.Text.TextChangeRange? range, System.Collections.Generic.IEnumerable<Microsoft.CodeAnalysis.Editor.DocumentSnapshotSpan> spansToCompute, System.Threading.CancellationToken cancellationToken) Line 68	C#
 	Microsoft.CodeAnalysis.EditorFeatures.dll!Microsoft.CodeAnalysis.Editor.Shared.Tagging.ProducerPopulatedTagSource<System.__Canon>.RecomputeTagsForeground.AnonymousMethod__0(System.Threading.CancellationToken ct) Line 372	C#
 	Microsoft.CodeAnalysis.EditorFeatures.dll!Microsoft.CodeAnalysis.Editor.Shared.Threading.AsynchronousSerialWorkQueue.EnqueueBackgroundTask.AnonymousMethod__0(System.Threading.Tasks.Task _) Line 130	C#
```

### PR DESCRIPTION
**Pull request by gafter**
Fixes #1099

### CODE REVIEW

**Comment by gafter** [comment]
@jaredpar @VSadov @AlekseyTs @agocke @VladimirReshetnikov
Could you review this please ASAP?

**Comment by jaredpar** [comment]
:+1:

**Comment by VSadov** [comment]
LGTM

File: src/Compilers/VisualBasic/Portable/Binding/Binder_Query.vb [hunk 1]
```diff
@@ -405,8 +405,7 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
 
                 If Not suppressCallDiagnostics AndAlso firstSelectExpression.HasErrors Then
                     Dim query = TryCast(firstSelectExpression, BoundQueryClause)
-
-                    suppressCallDiagnostics = (query.UnderlyingExpression.Kind = BoundKind.BadExpression)
+                    suppressCallDiagnostics = If(query?.UnderlyingExpression, firstSelectExpression).Kind = BoundKind.BadExpression
```
Comment (AlekseyTs) [inline src/Compilers/VisualBasic/Portable/Binding/Binder_Query.vb]: @gafter The UnderlyingExpression is never null, firstSelectExpression.Kind was checked before we entered this 'If', all we should check here is whether the query is Nothing. Please adjust the fix by simply putting additional 'If query IsNot Nothing' around original assignment.

**Pull request merged**